### PR TITLE
Support optional jobs

### PR DESCRIPTION
**Pull request by aliok**
Taking over https://github.com/openshift-knative/hack/pull/25

### CODE REVIEW

File: pkg/prowgen/prowgen_command_name_mapping.go [hunk 1]
```diff
@@ -15,23 +17,34 @@ const (
 // - name cannot be longer than maxNameLength characters.
 func ToName(r Repository, test *Test, openShiftVersion string) string {
 
+	name := test.Command
+	if test.OnDemand {
+		name = fmt.Sprintf("%s-ondemand", name)
+	}
```
[the remark this answers is not in the document]
Comment (aliok) [reply pkg/prowgen/prowgen_command_name_mapping.go:23]: That results in this:

```
 duplicate image name 'knative-eventing-kafka-broker-dispatcher' (previously defined by field 'images[0]')\n  * images[26]: duplicate image name 'knative-eventing-kafka-broker-receiver' (previously defined by field 'images[4]')\n 
```

[the remark this answers is not in the document]
Comment (pierDipi) [reply pkg/prowgen/prowgen_command_name_mapping.go:23]: Ah, that's my fault by it is unrelated to this change

[the remark this answers is not in the document]
Comment (pierDipi) [reply pkg/prowgen/prowgen_command_name_mapping.go:23]: working on a fix for that error

**Comment by aliok** [comment]
cc @pierDipi @creydr

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**